File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/ClientPlanSearchModel.ts
import { BaseSearchModel } from "./baseSearchModel";

export interface ClientPlanSearchModel extends BaseSearchModel {
    clientId: number | null;
    clientCode: string;

    branchId: number | null;
    branchCode: string;

    clientTypeId: number | null;
    
    fromDate: Date | null;
    toDate: Date | null;
}

File: SofiForce.Models/Models/SearchModels/ClientPlanSearchModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class ClientPlanSearchModel : BaseSearchModel
    {


        public int? ClientId { get; set; }
        public int? BranchId { get; set; }
        public int? ClientTypeId { get; set; }

        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
    }
}